SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/SubViewTest.cpp

```diff
@@ -0,0 +1,199 @@
+	return true;
+};
+TArray<FDispatcherRefreshCallback> NoRefreshCallbacks = TArray<FDispatcherRefreshCallback>{};
+
+SUBVIEW_TEST(GIVEN_Set_Of_Components_WHEN_Entity_Tagged_THEN_Components_Contains_Tag)
+{
+	FDispatcher Dispatcher;
+	EntityView View;
+	const FSubView SubView(TAG_COMPONENT_ID, NoFilter, &View, Dispatcher, NoRefreshCallbacks);
+	TArray<FWorkerComponentData> Components{ ComponentFactory::CreateEmptyComponentData(VALUE_COMPONENT_ID) };
+
+	SubView.TagEntity(Components);
+
+	TestTrue("Components contains tag component", Components.ContainsByPredicate([](const FWorkerComponentData& Data) {
+		return Data.component_id == TAG_COMPONENT_ID;
+	}));
+
+	return true;
+}
+
+SUBVIEW_TEST(GIVEN_Query_WHEN_Query_Tagged_THEN_Query_Is_Tagged)
+{
+	FDispatcher Dispatcher;
+	EntityView View;
+	FSubView SubView(TAG_COMPONENT_ID, NoFilter, &View, Dispatcher, NoRefreshCallbacks);
+	Query QueryToTag;
+	QueryConstraint Constraint;
+	Constraint.EntityIdConstraint = ENTITY_ID;
+	QueryToTag.Constraint = Constraint;
+	QueryToTag.ResultComponentIds = SchemaResultType{ VALUE_COMPONENT_ID };
+
+	SubView.TagQuery(QueryToTag);
+
+	if (!TestNotEqual("Constraint is now an AND", QueryToTag.Constraint.AndConstraint.Num(), 0))
+	{
+		return true;
+	}
+	TArray<QueryConstraint>& AndConstraint = QueryToTag.Constraint.AndConstraint;
+	Worker_ComponentId TestConstraintId = AndConstraint[0].ComponentConstraint.IsSet() ? AndConstraint[0].ComponentConstraint.GetValue()
+																					   : AndConstraint[1].ComponentConstraint.GetValue();
+	TestEqual("Constraint contains tag component constraint", TestConstraintId, TAG_COMPONENT_ID);
+
+	TestTrue("Result type contains tag", QueryToTag.ResultComponentIds.Contains(TAG_COMPONENT_ID));
+
+	return true;
+}
+
+SUBVIEW_TEST(GIVEN_SubView_Without_Filter_WHEN_Tagged_Entity_Added_THEN_Delta_Contains_Entity)
+{
+	FDispatcher Dispatcher;
+	EntityView View;
+	ViewDelta Delta;
+
+	FSubView SubView(TAG_COMPONENT_ID, NoFilter, &View, Dispatcher, NoRefreshCallbacks);
+
+	AddEntityToView(View, TAGGED_ENTITY_ID);
+	PopulateViewDeltaWithComponentAdded(Delta, View, TAGGED_ENTITY_ID, TAG_COMPONENT_ID);
+
+	Dispatcher.InvokeCallbacks(Delta.GetEntityDeltas());
+
+	SubView.Advance(Delta);
+	FSubViewDelta SubDelta = SubView.GetViewDelta();
+
+	// The tagged entity should pass through to the sub view delta.
+	if (!TestEqual("There is one entity delta", SubDelta.EntityDeltas.Num(), 1))
+	{
+		// early out so we don't crash - test has already failed
+		return true;
+	}
+	TestEqual("The entity delta is for the correct entity ID", SubDelta.EntityDeltas[0].EntityId, TAGGED_ENTITY_ID);
+
+	return true;
+}
+
+SUBVIEW_TEST(
+	GIVEN_SubView_With_Filter_WHEN_Tagged_Entities_Added_THEN_Delta_Only_Contains_Filtered_Entities_ALSO_Dispatcher_Callback_Refreshes_Correctly)
+{
+	FDispatcher Dispatcher;
+	EntityView View;
+	ViewDelta Delta;
+
+	FSubView SubView(
+		TAG_COMPONENT_ID,
+		[](const Worker_EntityId&, const EntityViewElement& Element) {
```

Comment:
nit - for readability can we forward define these input parameters

Reply:
code is now readable